src/components/contracts/ContractService.js

```diff
@@ -293,6 +294,20 @@ class ContractService {
     return event;
   };
 
+  defaultPeriod(contract) {
+    const startDate = getStartDateFromPeriod(getQuarterFromDate(contract.fieldValues.contract_start_date));
+    const endDate = getEndDateFromPeriod(getQuarterFromDate(contract.fieldValues.contract_end_date));
+    contract.fieldValues.contract_start_date = startDate;
```

Comment:
Better use a temp variable for contract: 
`const tempContract = { ...contract }` and return tempContract !